routing/index_graph_starter.cpp

```diff
@@ -176,9 +255,45 @@ void IndexGraphStarter::FindPointsWithCommonFeature(Joint::Id jointId0, Joint::I
   CHECK(found, ("Can't find common feature for joints", jointId0, jointId1));
 }
 
+void IndexGraphStarter::AddAdditionalZeroEdges(FakeJoint const & fp, EndPointType endPointType)
+{
+  CHECK(!m_graph.IsFakeFeature(fp.m_point.GetFeatureId()), ());
+
+  bool const isJoint = fp.BelongsToGraph();
+
+  vector<DirectedEdge> edges;
+  if (isJoint)
+  {
+    m_graph.GetEdgeList(fp.m_jointId, endPointType == EndPointType::Start /* is outgoing */,
+                        true /* graphWithoutRestrictions */, edges);
+  }
+  else
+  {
+    m_graph.GetIntermediatePointEdges(fp.m_point, true /* graphWithoutRestrictions */, edges);
+  }
+
+  for (DirectedEdge const & edge : edges)
+  {
+    m_graph.ForEachEdgeMappingNode(edge, [&](DirectedEdge const & e){
+      if (edge == e)
```

Comment:
In case of start joint in this ForEach...() we connect one of ends (joints) of source edge (|edge|) with all its clones(|e|). If to remove this if we connect one ends (joints) of |edge| with itself with a zero length edge. I guess it's redundant.